src/content/1.7/testing/how-to-create-your-own-symfony-controller-tests.md

```diff
@@ -0,0 +1,43 @@
+---
+title: How to create your own symfony controller tests
+menuTitle: Creating your own symfony controller tests
+weight: 20
+---
+
+# How to create your own symfony controller tests or add tests to PrestaShop
+
+If you create your own Modern (symfony) controllers or if you plan on migrating a legacy controller, you can create
+tests to cover parts of HTTP layer aka symfony functional tests.
+
+## Creating a functional test
+
+To create a functional test, we encourage you to base your test class on our own implementation of `WebTestCase`.
+
+For instance:
+
+```php
+namespace Tests\Functional\Foo;
+
+use Tests\Integration\PrestaShopBundle\Test\WebTestCase;
+
+class BarTest extends WebTestCase
+{
+    public function testSomeAction()
+    {
+        $url = '/modules/your/action';
+        /** or using the router
```

Comment:
Better to show the `router->generate` only.
Prevent a BC in case they don't use the router and we change the URL.